config/smarty.config.inc.php

```diff
@@ -177,9 +177,7 @@ function smartyCleanHtml($data)
 function smartyClassname($classname)
 {
     $classname = Tools::replaceAccentedChars(strtolower($classname));
-    $classname = preg_replace('/[^A-Za-z0-9]/', '-', $classname);
-    $classname = preg_replace('/[-]+/', '-', $classname);
-
+    $classname = preg_replace(array("/[^A-Za-z0-9-_]/", "/[-]{3,}/", "/[-]+$/"), array("-", '-', '') , $classname);
```

Comment:
```suggestion
    $classname = preg_replace(['/[^A-Za-z0-9-_]/', '/-{3,}/', '/-+$/'], ['-', '-', ''] , $classname);
```
Characters group are not needed when there is no group :sweat_smile: